# Análise Básica do Dataset (Movies)
Este notebook carrega o arquivo `movies.csv` e executa análises básicas solicitadas: visualização de amostras, dimensões, tipos, estatísticas descritivas, nomes de colunas e valores ausentes.

In [9]:
# Importar bibliotecas e carregar o CSV
import pandas as pd

# Caminho do arquivo (use r'...' para lidar com caracteres especiais)
caminho = r'C:\Users\rafae\OneDrive\Desktop\Fatesg-IA-2\Ciência de dados\Atividade Manipulação Básica de Dados\movies.csv'

# Carregar dados
df = pd.read_csv(caminho)

# Visualizar forma geral
print('Dimensões iniciais -> Linhas:', df.shape[0], '| Colunas:', df.shape[1])

Dimensões iniciais -> Linhas: 4803 | Colunas: 24


In [10]:
# 2. Exiba as 8 primeiras e 8 últimas linhas
from IPython.display import display

print('8 primeiras linhas:')
display(df.head(8))
print('\n8 últimas linhas:')
display(df.tail(8))

8 primeiras linhas:


index     budget                                    genres  \
0      0  237000000  Action Adventure Fantasy Science Fiction   
1      1  300000000                  Adventure Fantasy Action   
2      2  245000000                    Action Adventure Crime   
3      3  250000000               Action Crime Drama Thriller   
4      4  260000000          Action Adventure Science Fiction   
5      5  258000000                  Fantasy Action Adventure   
6      6  260000000                          Animation Family   
7      7  280000000          Action Adventure Science Fiction   

                                            homepage      id  \
0                        http://www.avatarmovie.com/   19995   
1       http://disney.go.com/disneypictures/pirates/     285   
2        http://www.sonypictures.com/movies/spectre/  206647   
3                 http://www.thedarkknightrises.com/   49026   
4               http://movies.disney.com/john-carter   49529   
5    http://www.sonypictures.com/movies/spider-man3/     559   
6       http://disney.go.com/disneypictures/tangled/   38757   
7  http://marvel.com/movies/movie/193/avengers_ag...   99861   

                                            keywords original_language  \
0  culture clash future space war space colony so...                en   
1  ocean drug abuse exotic island east india trad...                en   
2         spy based on novel secret agent sequel mi6                en   
3  dc comics crime fighter terrorist secret ident...                en   
4  based on novel mars medallion space travel pri...                en   
5  dual identity amnesia sandstorm love of one's ...                en   
6             hostage magic horse fairy tale musical                en   
7  marvel comic sequel superhero based on comic b...                en   

                             original_title  \
0                                    Avatar   
1  Pirates of the Caribbean: At World's End   
2                                   Spectre   
3                     The Dark Knight Rises   
4                               John Carter   
5                              Spider-Man 3   
6                                   Tangled   
7                   Avengers: Age of Ultron   

                                            overview  popularity  ... runtime  \
0  In the 22nd century, a paraplegic Marine is di...  150.437577  ...   162.0   
1  Captain Barbossa, long believed to be dead, ha...  139.082615  ...   169.0   
2  A cryptic message from Bond’s past sends him o...  107.376788  ...   148.0   
3  Following the death of District Attorney Harve...  112.312950  ...   165.0   
4  John Carter is a war-weary, former military ca...   43.926995  ...   132.0   
5  The seemingly invincible Spider-Man goes up ag...  115.699814  ...   139.0   
6  When the kingdom's most wanted-and most charmi...   48.681969  ...   100.0   
7  When Tony Stark tries to jumpstart a dormant p...  134.279229  ...   141.0   

                                    spoken_languages    status  \
0  [{"iso_639_1": "en", "name": "English"}, {"iso...  Released   
1           [{"iso_639_1": "en", "name": "English"}]  Released   
2  [{"iso_639_1": "fr", "name": "Fran\u00e7ais"},...  Released   
3           [{"iso_639_1": "en", "name": "English"}]  Released   
4           [{"iso_639_1": "en", "name": "English"}]  Released   
5  [{"iso_639_1": "en", "name": "English"}, {"iso...  Released   
6           [{"iso_639_1": "en", "name": "English"}]  Released   
7           [{"iso_639_1": "en", "name": "English"}]  Released   

                                          tagline  \
0                     Enter the World of Pandora.   
1  At the end of the world, the adventure begins.   
2                           A Plan No One Escapes   
3                                 The Legend Ends   
4            Lost in our world, found in another.   
5                              The battle within.   
6        They're taking adventure to new lengths.   



8 últimas linhas:


index  budget                          genres  \
4795   4795       0                           Drama   
4796   4796    7000  Science Fiction Drama Thriller   
4797   4797       0                Foreign Thriller   
4798   4798  220000           Action Crime Thriller   
4799   4799    9000                  Comedy Romance   
4800   4800       0   Comedy Drama Romance TV Movie   
4801   4801       0                             NaN   
4802   4802       0                     Documentary   

                                               homepage      id  \
4795                                                NaN  124606   
4796                         http://www.primermovie.com   14337   
4797                                                NaN   67238   
4798                                                NaN    9367   
4799                                                NaN   72766   
4800  http://www.hallmarkchannel.com/signedsealeddel...  231617   
4801                        http://shanghaicalling.com/  126186   
4802                                                NaN   25975   

                                               keywords original_language  \
4795         gang audition police fake homeless actress                en   
4796  distrust garage identity crisis time travel ti...                en   
4797                                                NaN                en   
4798  united states\u2013mexico barrier legs arms pa...                es   
4799                                                NaN                en   
4800  date love at first sight narration investigati...                en   
4801                                                NaN                en   
4802               obsession camcorder crush dream girl                en   

                 original_title  \
4795                       Bang   
4796                     Primer   
4797                     Cavite   
4798                El Mariachi   
4799                  Newlyweds   
4800  Signed, Sealed, Delivered   
4801           Shanghai Calling   
4802          My Date with Drew   

                                               overview  popularity  ...  \
4795  A young woman in L.A. is having a bad day: she...    0.918116  ...   
4796  Friends/fledgling entrepreneurs invent a devic...   23.307949  ...   
4797  Adam, a security guard, travels from Californi...    0.022173  ...   
4798  El Mariachi just wants to play his guitar and ...   14.269792  ...   
4799  A newlywed couple's honeymoon is upended by th...    0.642552  ...   
4800  "Signed, Sealed, Delivered" introduces a dedic...    1.444476  ...   
4801  When ambitious New York attorney Sam is sent t...    0.857008  ...   
4802  Ever since the second grade when he first saw ...    1.929883  ...   

     runtime                               spoken_languages    status  \
4795    98.0       [{"iso_639_1": "en", "name": "English"}]  Released   
4796    77.0       [{"iso_639_1": "en", "name": "English"}]  Released   
4797    80.0                                             []  Released   
4798    81.0  [{"iso_639_1": "es", "name": "Espa\u00f1ol"}]  Released   
4799    85.0                                             []  Released   
4800   120.0       [{"iso_639_1": "en", "name": "English"}]  Released   
4801    98.0       [{"iso_639_1": "en", "name": "English"}]  Released   
4802    90.0       [{"iso_639_1": "en", "name": "English"}]  Released   

                                                tagline  \
4795            Sometimes you've got to break the rules   
4796                 What happens if it actually works?   
4797                                                NaN   
4798  He didn't come looking for trouble, but troubl...   
4799  A newlywed couple's honeymoon is upended by th...   
4800                                                NaN   
4801                           A New Yorker in Shanghai   
4802                                                NaN   

                     

In [11]:
# 3. Número total de linhas e colunas
print(f"Linhas: {df.shape[0]}")
print(f"Colunas: {df.shape[1]}")

Linhas: 4803
Colunas: 24


In [12]:
# 4. Tipos de cada coluna
# df.info() imprime diretamente o resumo das colunas
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 4803 entries, 0 to 4802
Data columns (total 24 columns):
 #   Column                Non-Null Count  Dtype  
---  ------                --------------  -----  
 0   index                 4803 non-null   int64  
 1   budget                4803 non-null   int64  
 2   genres                4775 non-null   object 
 3   homepage              1712 non-null   object 
 4   id                    4803 non-null   int64  
 5   keywords              4391 non-null   object 
 6   original_language     4803 non-null   object 
 7   original_title        4803 non-null   object 
 8   overview              4800 non-null   object 
 9   popularity            4803 non-null   float64
 10  production_companies  4803 non-null   object 
 11  production_countries  4803 non-null   object 
 12  release_date          4802 non-null   object 
 13  revenue               4803 non-null   int64  
 14  runtime               4801 non-null   float64
 15  spoken_languages     

In [13]:
# 5a. Estatísticas descritivas (apenas colunas numéricas por padrão)
df.describe()

index        budget             id   popularity       revenue  \
count  4803.000000  4.803000e+03    4803.000000  4803.000000  4.803000e+03   
mean   2401.000000  2.904504e+07   57165.484281    21.492301  8.226064e+07   
std    1386.651002  4.072239e+07   88694.614033    31.816650  1.628571e+08   
min       0.000000  0.000000e+00       5.000000     0.000000  0.000000e+00   
25%    1200.500000  7.900000e+05    9014.500000     4.668070  0.000000e+00   
50%    2401.000000  1.500000e+07   14629.000000    12.921594  1.917000e+07   
75%    3601.500000  4.000000e+07   58610.500000    28.313505  9.291719e+07   
max    4802.000000  3.800000e+08  459488.000000   875.581305  2.787965e+09   

           runtime  vote_average    vote_count  
count  4801.000000   4803.000000   4803.000000  
mean    106.875859      6.092172    690.217989  
std      22.611935      1.194612   1234.585891  
min       0.000000      0.000000      0.000000  
25%      94.000000      5.600000     54.000000  
50%     103.000000      6.200000    235.000000  
75%     118.000000      6.800000    737.000000  
max     338.000000     10.000000  13752.000000

In [14]:
# 5b. Nomes das colunas
list(df.columns)

['index',
 'budget',
 'genres',
 'homepage',
 'id',
 'keywords',
 'original_language',
 'original_title',
 'overview',
 'popularity',
 'production_companies',
 'production_countries',
 'release_date',
 'revenue',
 'runtime',
 'spoken_languages',
 'status',
 'tagline',
 'title',
 'vote_average',
 'vote_count',
 'cast',
 'crew',
 'director']

In [15]:
# 5c. Valores ausentes em cada coluna
# Dica: use .sort_values(ascending=False) para ordenar do maior para o menor
df.isnull().sum().sort_values(ascending=False)

homepage                3091
tagline                  844
keywords                 412
cast                      43
director                  30
genres                    28
overview                   3
runtime                    2
release_date               1
original_title             0
id                         0
index                      0
budget                     0
original_language          0
production_countries       0
popularity                 0
spoken_languages           0
revenue                    0
production_companies       0
status                     0
vote_average               0
title                      0
vote_count                 0
crew                       0
dtype: int64

# 5d. Impressões gerais sobre o dataset

Com base apenas na observação das características do arquivo:

- Tamanho: o dataset possui 4803 linhas e 24 colunas, sugerindo boa quantidade de exemplos para análise exploratória.
- Tipos de dados: há mistura de variáveis numéricas (por exemplo, `budget`, `revenue`, `popularity`, `vote_average`, `vote_count`) e categóricas/textuais (por exemplo, `genres`, `original_language`, `status`, `title`, `overview`). A coluna `release_date` aparece como texto e pode ser convertida para data.
- Valores ausentes: maior incidência em `homepage` (muitos filmes não possuem página oficial) e `tagline`; também há ausências em `keywords`, `cast`, `director`, `genres`. Colunas numéricas têm poucos ou nenhum nulo.
- Possíveis análises: relação entre `budget` e `revenue`; distribuição de `vote_average` e `runtime`; popularidade por gênero e por idioma; variações temporais após converter `release_date` para data (ano/mês).
- Pré-processamento: recomenda-se padronizar tipos (datas), tratar ausentes (remover, imputar ou manter conforme objetivo), e eventualmente expandir colunas aninhadas (como `genres`, `production_companies`) se forem listas/strings estruturadas.

Essas impressões são iniciais e servem para orientar próximos passos de limpeza e exploração.